org.eclipse.languageserver/src/org/eclipse/languageserver/LSPEclipseUtils.java

```diff
@@ -91,4 +91,37 @@ public static IResource findResourceFor(String uri) {
 		return resource;
 	}
 
+	public static class LSPDocumentInfo {
```

Comment:
I would rather see this method in the LanguageServiceAccessor and the LanguageServiceAccessor.getLanguageServer() made private then. WDYT?